pkg/kubelet/dockershim/docker_sandbox_test.go

```diff
@@ -154,6 +158,33 @@ func TestSandboxStatus(t *testing.T) {
 	assert.Error(t, err, fmt.Sprintf("status of sandbox: %+v", statusResp))
 }
 
+// TestSandboxHasLeastPrivilegesConfig tests that the sandbox is set with no-new-privileges
+// and it has a seccomp profile.
+func TestSandboxHasLeastPrivilegesConfig(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.PodSandboxSeccomp, true)()
+
+	ds, _, _ := newTestDockerService()
+	config := makeSandboxConfig("foo", "bar", "1", 0)
+	ds.podSandBoxSeccomp = `{defaultAction="SCMP_ACT_ERRNO"}`
+
+	// test the default
+	createConfig, err := ds.makeSandboxDockerConfig(config, defaultSandboxImage)
+	assert.NoError(t, err)
+	assert.Equal(t, "no-new-privileges", createConfig.HostConfig.SecurityOpt[0], "no-new-privileges not set")
+	assert.True(t, strings.HasPrefix(createConfig.HostConfig.SecurityOpt[1], `seccomp={defaultAction="SCMP_ACT_ERRNO"}`), "seccomp not set")
+
+	// test that users can't overwrite the sandboxes' seccomp
+	config.Linux = &runtimeapi.LinuxPodSandboxConfig{
```

Comment:
override no-new-privs too?

Reply:
@tallclair yes, it needs to in order to decrease the amount of required permissions. The runtimes tend to have a small permission requirement when executed with 'no-new-privs' on.